core/ev-admin.php

```diff
@@ -0,0 +1,1049 @@
+  *  @since  1.0
+  *
+  *  @param
+  *  @return
+  */
+
+  function plugin_settings_handler() {
+
+    // Handle the ajax request
+    check_ajax_referer( 'ev_settings' );
+
+    $options = SP_External_Videos::get_options();
+    $message = '';
+
+    $fields = array();
+    parse_str( $_POST['data'], $fields );
+
+    // error_log( print_r( $fields, 1 ) );
+
+    $options['rss'] = ( array_key_exists( 'ev-rss', $fields ) ? true : false );
+    $options['delete'] = ( array_key_exists( 'ev-delete', $fields ) ? true : false );
+    $options['attrib'] = ( array_key_exists( 'ev-attrib', $fields ) ? true : false );
+    $options['loop'] = ( array_key_exists( 'ev-loop', $fields ) ? true : false );
+
+    if( update_option( 'sp_external_videos_options', $options ) ) $message = "Settings saved.";
+    ob_clean();
+
+    wp_send_json( $message );
+
+  }
+
+  /*
+  *  update_videos_handler
+  *
+  *  Used by settings page
+  *  AJAX handler for "Update videos from channels" form
+  *
+  *  @type  function
+  *  @date  31/10/16
+  *  @since  1.0
+  *
+  *  @param
+  *  @return
+  */
+
+  function update_videos_handler() {
+
+    // Handle the ajax request
+    check_ajax_referer( 'ev_settings' );
+
+    $options = SP_External_Videos::get_options();
+    $AUTHORS = SP_EV_Admin::admin_get_authors();
+    $messages = '';
+
+    if( isset( $_POST['author_id'] ) && isset( $_POST['host_id'] ) ) {
+      $eligibles = array();
+
+      foreach( $AUTHORS as $authorkey => $authordata ){
+        if( in_array( $_POST['host_id'], $authordata ) ) {
+          $eligibles[] = $authorkey;
+        }
+      }
+
+      if( $eligibles ) {
+        $thisindex = array();
+
+        foreach( $eligibles as $eligible => $originalkey ){
+          if( in_array( $_POST['author_id'], $AUTHORS[$originalkey] ) ) {
+            $thisindex[] = $originalkey;
+          }
+        }
+        $thisindex = array_shift( $thisindex );
+        $update_authors = array( $AUTHORS[$thisindex] );
+        $single = true;
+      }
+
+    } else {
+      $update_authors = $AUTHORS;
+      $single = null;
+    }
+
+    $delete = isset( $options['delete'] ) ? $options['delete'] : null;
+    // post_new_videos() gets everything new or deleted and returns messages about it
+    $messages = $this->post_new_videos( $update_authors, $delete, $single );
+
+    wp_send_json( $messages );
+
+  }
+
+  /*
+  *  post_new_videos
+  *
+  *  Used by update_videos_handler() and daily_function()
+  *  Saves any new videos from host channels to the database.
+  *  Returns number of video posts added.
+  *  Works for single-author and update-all via $single param
+  *
+  *  @type  function
+  *  @date  31/10/16
+  *  @since  1.0
+  *
+  *  @param   $authors, $delete, $single
+  *  @return  html $messages
+  */
+
+  function post_new_videos( $authors, $delete, $single ) {
+
+    $HOSTS = SP_EV_Admin::admin_get_hosts();
+
+    $new_videos = $this->fetch_new_videos( $authors, $HOSTS );
+
+    if ( !$new_videos ) return 0;
```

Comment:
In ev-vimeo.php, you might want look at what returns in $body and do something like this:
if ( $body['error'] ) {
  throw new Exception($body['error']);
}